prow/cmd/deck/github_oauth_setup.md

```diff
@@ -0,0 +1,119 @@
+1. Create your GitHub Oauth application 
+
+    https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/
+    
+    Make sure to create a GitHub Oauth App and not a regular GitHub App.
+    
+    The callback url should be:
+    
+    `<PROW_BASE_URL>/github-login/redirect`
+2. Create a secret file for GitHub OAuth that has the following content. The information can be found in the [GitHub OAuth developer settings](https://github.com/settings/developers):
+    
+    ```
+    client_id: <APP_CLIENT_ID>
+    client_secret: <APP_CLIENT_SECRET>
+    redirect_url: <PROW_BASE_URL>/github-login/redirect
+    final_redirect_url: <PROW_BASE_URL>/pr
+    ```
+    
+    If Prow is expected to work with private repositories, add
+    ```
+    scopes:
+    - repo
+    ```
+    
+3. Create another secret file for the cookie store. This cookie secret will also be used for CSRF protection.
+  The file should contain a random 32-byte length base64 key. For example, you can use `openssl` to generate the key
+    
+    ```
+    openssl rand -out cookie.txt -base64 32
+    ```
+4. Use `kubectl`, which should already point to your Prow cluster, to create secrets using the command:
+    
+    `kubectl create secret generic github-oauth-config --from-file=secret=<PATH_TO_YOUR_GITHUB_SECRET>`
+
+    `kubectl create secret generic cookie --from-file=secret=<PATH_TO_YOUR_COOKIE_KEY_SECRET>`
+5. To use the secrets, you can either:
+
+    * [Mount](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets) secrets to your deck volume:
+
+        Open `test-infra/prow/cluster/deck_deployment.yaml`.
+        Under `volumes` token, add:
+        ```
+        - name: oauth-config
+          secret:
+              secretName: github-oauth-config
+        - name: cookie-secret
+          secret:
+              secretName: cookie
+        ```
+        Under `volumeMounts` token, add:
+        ```
+        - name: oauth-config
+          mountPath: /etc/githuboauth
+          readOnly: true
+        - name: cookie-secret
+          mountPath: /etc/cookie
+          readOnly: true
+        ```
+    * Add the following flags to `deck`:
+      ```
+      - --github-oauth-config-file=/etc/githuboauth/secret
+      - --oauth-url=/github-login
+      ```
+      Note that the `--oauth-url` should eventually be changed to a boolean as described 
+      in [#13804](https://github.com/kubernetes/test-infra/issues/13804).
+    * You can also set your own path to the cookie secret using the `--cookie-secret` flag.
+    * To prevent `deck` from making mutating GitHub API calls, pass in the `--dry-run` flag.
+
+##Configuring GitHub bot
+The rerun button can be configured so that certain GitHub teams are allowed to trigger certain jobs
+from the frontend. In order to make API calls to determine whether a user is on a given team, `deck` needs 
+to use the access token of an org member. If you have a bot already set up, we recommend using its token,
```

Comment:
Not sure I agree with this recommendation, given that deck is exposed to the public it is most likely safer to use a separate token that does not have write access to the repo. Deck should use a different token from tide and hook.